server/src/main/java/org/opensearch/transport/BaseInboundMessage.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.transport;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Base class for inbound data as a message.
+ * Different implementations are used for different protocols.
+ *
+ * @opensearch.internal
+ */
+@ExperimentalApi
+public interface BaseInboundMessage {
+
+    /**
+     * The protocol used to encode this message
+     */
+    static String NATIVE_PROTOCOL = "native";
+    static String PROTOBUF_PROTOCOL = "protobuf";
+
+    /**
+     * @return the protocol used to encode this message
+     */
+    public String getProtocol();
+
+    /**
+     * Set the protocol used to encode this message
+     */
+    public void setProtocol();
```

Comment:
What are the use-cases for setting protocol? I would only include `getProtocol()`.

Reply:
The implementation classes will ultimately set the protocol, the change is not in this PR but it will be used for protobuf like [this](https://github.com/opensearch-project/OpenSearch/pull/11910/files#diff-6112eb58e35917cfcadec0918d1ae5925beacc4fca9e1eb0cc90c50b1f3a2e57R138).

Reply:
I believe the message is always constructed per protocol - there is no need to set it explicitly